package/MDAnalysis/analysis/base.py

```diff
@@ -40,20 +41,118 @@
 logger = logging.getLogger(__name__)
 
 
+class Results(dict):
+    r"""Container object for storing results.
+
+    Results are dictionaries that provide two ways in which can values be
+    accessed: by dict key ``results["value_key"]`` or by object attribute,
+    ``results.value_key``. They store all results obatined from an analysis
+    after the ``run`` call.
+
+    The current implementation is similar to the :class:`sklearn.utils.Bunch`
+    class in `scikit-learn`_.
+
+    .. _`scikit-learn`: https://scikit-learn.org/
+
+
+    Examples
+    --------
+    >>> from MDAnalysis.analysis.base import Results
+    >>> results = Results(a=1, b=2)
+    >>> results['b']
+    2
+    >>> results.b
+    2
+    >>> results.a = 3
+    >>> results['a']
+    3
+    >>> results.c = [1, 2, 3, 4]
+    >>> results['c']
+    [1, 2, 3, 4]
+
+
+    Raises
+    ------
+    TypeError
+        If an attribute would have the same name as a default dictionary
+        attribute.
+
+    TypeError
+        If a key is not of type ``str`` and therefore is not able to be
+        accessed by attribute.
+    """
+
+    def _validate_key(self, key):
+        if key in dir(dict):
+            raise TypeError(f"'{key}' is a protected dictionary attribute")
+        elif not (isinstance(key, str) and key[0].isalpha()
+                  and re.match("^[a-zA-Z0-9_]*$", key)):
```

Comment:
What does the regex do that `key[0].isalpha()` doesn't?

Reply:
I imagine you'd also run into trouble if people sprinkled "~*=-" etc. after the first character.

Reply:
Can you combine the isalpha into the regex with ``re.match("^[a-zA-Z_][a-zA-Z0-9_]*$")`` ?

Reply:
Yes, good idea. But I think I got some errors with spaces...